pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -254,6 +254,59 @@ func getAllDomainDisks(dom cli.VirDomain) ([]api.Disk, error) {
 	return devices.Disks, nil
 }
 
+func (l *LibvirtDomainManager) hotplugMemory(vmi *v1.VirtualMachineInstance, spec *api.DomainSpec, dom cli.VirDomain) error {
+
+	// check whether virt-launcher is ready for hotplug
+	if _, exists := vmi.Labels[v1.MemoryHotplugReadyLabel]; !exists {
+		return nil
+	}
+
+	if vmi.Spec.Domain.Memory == nil ||
+		vmi.Spec.Domain.Memory.Guest == nil ||
+		vmi.Spec.Domain.Memory.MaxGuest == nil ||
+		vmi.Status.Memory == nil ||
+		vmi.Status.Memory.GuestCurrent == nil ||
+		vmi.Status.Memory.GuestAtBoot == nil {
+		return nil
+	}
+
+	if vmi.Status.Memory.GuestCurrent.Cmp(*vmi.Spec.Domain.Memory.Guest) == 0 {
```

Comment:
Could that be a `GuestCurrent` is lacking behind? Wouldn't it be more resilient to consult `domain`? In other words, what does this give us that the below check is not giving us?

Reply:
It should be eventually consistent. This check tells you that hotplug was requested, the one below makes sure that the request has already been processed. Am I missing something?